### PR TITLE
Make ReactiveClientHeadersFactory replace headers instead of adding - backport to 2.7

### PR DESCRIPTION
**Pull request by michalszynkiewicz**

### CODE REVIEW

**Comment by michalszynkiewicz** [comment]
the failures are not related

File: extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/ReactiveClientHeadersFactory.java [hunk 1]
```diff
@@ -10,7 +10,35 @@
...
+     *
+     * @see ClientHeadersFactory#update(MultivaluedMap, MultivaluedMap)
+     */
+    public Uni<MultivaluedMap<String, String>> getHeaders(MultivaluedMap<String, String> incomingHeaders,
+            MultivaluedMap<String, String> clientOutgoingHeaders) {
+        return getHeaders(incomingHeaders);
+    }
+
+    /**
+     * @deprecated Will be removed in Quarkus 2.8. Use
```
Comment (gsmet) [inline extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/ReactiveClientHeadersFactory.java:31]:
```suggestion
     * @deprecated Will be removed in Quarkus 2.8. Implement and use
```

File: extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/ReactiveClientHeadersFactory.java [hunk 2]
```diff
@@ -10,7 +10,35 @@
...
+     *             {@link ReactiveClientHeadersFactory#getHeaders(MultivaluedMap, MultivaluedMap)} instead
+     *
+     * @param incomingHeaders the map of headers from the inbound JAX-RS request. This will be an empty map if the
+     *        associated client interface is not part of a JAX-RS request.
+     * @return a Uni with a map of HTTP headers to merge with the clientOutgoingHeaders to be sent to the remote service.
+     */
+    @Deprecated
+    public Uni<MultivaluedMap<String, String>> getHeaders(MultivaluedMap<String, String> incomingHeaders) {
+        throw new IllegalStateException(getClass() + " does not implement neither of the getHeaders() methods");
```
Comment (gsmet) [inline extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/ReactiveClientHeadersFactory.java:40]:
```suggestion
        throw new IllegalStateException(getClass() + " does not implement either of the getHeaders() methods");
```

**Review by gsmet — CHANGES_REQUESTED** [review]
I will apply the two wording changes and merge as CI looks OK.

**Pull request merged**